test/python/circuit/test_registerless_circuit.py

```diff
@@ -43,6 +44,61 @@ def test_circuit_constructor_wires_wrong_mix(self):
         self.assertRaises(CircuitError, QuantumCircuit, 1, ClassicalRegister(2))
 
 
+class TestAddingBitsWithoutRegisters(QiskitTestCase):
```

Comment:
The only test I would think to add here is one for the mixed register, registerless bit case that index in gates works as expected. But that might be in one of the follow on commits, I haven't looked at those yet.

Reply:
Good point. I don't think there's a test covering mixed regisered and registerless Bits in the tests covering addition of gates on a specific qubit index (updated in #5498). I'll add one there.

Reply:
Test added in https://github.com/Qiskit/qiskit-terra/pull/5498/commits/71ec0a451147e4b2e53935039d10fa4397f54fbb .